OpenRA.Mods.Common/Traits/Sound/AmbientSound.cs

```diff
@@ -70,5 +81,13 @@ public void Tick(Actor self)
 			else
 				Game.Sound.Play(Info.SoundFile);
 		}
+
+		void INotifyRemovedFromWorld.RemovedFromWorld(Actor self)
+		{
+			Game.Sound.StopSound(currentSound);
```

Comment:
Please pull out a common `StopSound` method instead of duplicating this here and in Tick.